src/transformers/tools/agents.py

```diff
@@ -121,658 +122,369 @@ def _setup_default_tools():
+        """
+        This template is taking can desbribe a tool as it is expected by the model
+        """
+        logger.warning_once(
+            "\nNo tool description template is defined for this tokenizer - using a default tool description template "
+            "that implements the ChatML format (without BOS/EOS tokens!). If the default is not appropriate for "
+            "your model, please set `tokenizer.tool_description_template` to an appropriate template. "
         )
+        return DEFAULT_TOOL_DESCRIPTION_TEMPLATE
+    
 
-    @classmethod
-    def from_pretrained(cls, pretrained_model_name_or_path, **kwargs):
+    def run(self, task):
         """
-        Convenience method to build a `LocalAgent` from a pretrained checkpoint.
+        Sends a request to the agent.
 
         Args:
-            pretrained_model_name_or_path (`str` or `os.PathLike`):
-                The name of a repo on the Hub or a local path to a folder containing both model and tokenizer.
-            kwargs (`Dict[str, Any]`, *optional*):
-                Keyword arguments passed along to [`~PreTrainedModel.from_pretrained`].
+            task (`str`): The task to perform
 
         Example:
 
         ```py
-        import torch
-        from transformers import LocalAgent
+        from transformers import ReactAgent
+        from transformers.tools.base import CalculatorTool
 
-        agent = LocalAgent.from_pretrained("bigcode/starcoder", device_map="auto", torch_dtype=torch.bfloat16)
-        agent.run("Draw me a picture of rivers and lakes.")
+        calculator = CalculatorTool()
+        agent = ReactAgent(toolbox=[CalculatorTool()])
+        agent.run("What is the result of 2 power 3.7384?")
         ```
         """
-        model = AutoModelForCausalLM.from_pretrained(pretrained_model_name_or_path, **kwargs)
-        tokenizer = AutoTokenizer.from_pretrained(pretrained_model_name_or_path, **kwargs)
-        return cls(model, tokenizer)
-
-    @property
-    def _model_device(self):
-        if hasattr(self.model, "hf_device_map"):
-            return list(self.model.hf_device_map.values())[0]
-        for param in self.model.parameters():
-            return param.device
-
-    def generate_one(self, prompt, stop):
-        encoded_inputs = self.tokenizer(prompt, return_tensors="pt").to(self._model_device)
-        src_len = encoded_inputs["input_ids"].shape[1]
-        stopping_criteria = StoppingCriteriaList([StopSequenceCriteria(stop, self.tokenizer)])
-        outputs = self.model.generate(
-            encoded_inputs["input_ids"], max_new_tokens=200, stopping_criteria=stopping_criteria
-        )
-
-        result = self.tokenizer.decode(outputs[0].tolist()[src_len:])
-        # Inference API returns the stop sequence
-        for stop_seq in stop:
-            if result.endswith(stop_seq):
-                result = result[: -len(stop_seq)]
-        return result
-
+        self.memory = [self.system_prompt]
```

Comment:
I think it's good if it's public!

Reply:
It's already created at instance initialization in `Agent`: the idea here was that `run` should reset the agent's memory, since the philosophy is "1 agent = 1 task", and `run()` should solve the task at hand.